indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -365,6 +365,7 @@ public TaskStatus call()
                                   )
                               );
 
+                              command.add(String.format("-Ddruid.indexer.taskId=%s", task.getId()));
```

Comment:
@drcrallen hm this might not actually be all that straightforward - I need access to the taskId in ChatHandlerServerModule.configure() so that I can create the binding for the response header servlet filter and at this stage the injector hasn't been created yet. What would you recommend as the best way to get access to DataSourceTaskIdHolder from here?